### DISCUSSION TITLE
Problem on running apk downloading from whatsapp or google drive, says app not installed, adb and cordova run works, its just an helloworld

### DISCUSSION THREAD

**Discussion by Sebastiangperez** [apache/cordova#419] [General]
Question , i have a problem with a build , i made a apk file , i signed , i can test it using cordova run with --device flag and works also using adb cli command , but if i want to install from whatsapp or using google drive or sending to the phone with a file manager , said app not installed , do you know what could be the problem ?
i'm using a S10 phone , android 12 , and the app is using the latest sdk , but its just an helloworld.

**Comment by Sebastiangperez** [comment]
I know what the problem is , Play Protect , if you disable it , works, but wont work if you press install anyway.

**Comment by breautek** [comment]
This is standard Android OS behaviour.

By default, only a select few apps can install apps on the device, which is usually Google Play Store and generally the phone manufacturer's app store (e.g. Samsung Galaxy Store).

Apps can also be installed to developer devices (phones with USB debugging enabled) via ADB, which is the channel that Cordova uses to install apps onto the device. This is something that needs to be explicitly enabled by the user.

If you manually transfer the APK  to your device, you must allow the source app that accepted the APK to install APKs on the device. This can be done by going to `Settings -> Security -> Install unknown apps`.

On older android versions, install unknown apps is a simple target that was disabled by default, and enabling it will allow you to install APks from untrusted sources. Newer android versions will go into a subscreen where you enable access for specific apps to allow APKs to be installed from specific sources. E.g. if you download an APK from Google Drive, then you must enable `Allow from this source` on the Google Drive app to install APKs obtained from the Drive app.

None of these methods is a recommended way of publishing your app if you can help it. Your end users must jump through a lot of security hoops to install your app this fashion. The play stores often have a certain standard that your app must uphere too, which indirectly gives users some sense of trust when they install your app through official play stores.

**Reply by Sebastiangperez** [reply]
Thanks for your reply , when you said this "if you download an APK from Google Drive, then you must enable Allow from this source on the Google Drive app to install APKs obtained from the Drive app." I did that but also wont let me , i mean , said " the app cannot be installed " but if i disable Play Protect , i can. this change anything? i mean , the app must be installed anyway?  
This app will be upload to Google Store so , i think won't be a problem no ?

**Reply by breautek** [reply]
Google Play Protect is something rather new so I'm not really familiar with the ends and outs of it. But it is software that acts kind of like an antivirus, and if it thinks your app is not trustworthy then it may block it I suppose.

https://support.google.com/googleplay/answer/2812853?hl=en

So Google Play Protect is a secondary layer of security on top of allowing untrusted APKs.

**Reply by Sebastiangperez** [reply]
Thanks @breautek for your info , i will check when i upload my apk or aab to Google Play.
